source/core/Variants.cpp

```diff
@@ -75,23 +76,32 @@ VIREO_FUNCTION_SIGNATURET(VariantToData, VariantToDataParamBlock)
         TypeRef inputType = _ParamImmediate(InputData._paramType);
         void* inputData = _ParamImmediate(InputData)._pData;
 
-        TypeRef destType = _ParamImmediate(DestData._paramType);
+        TypeRef destType = _ParamImmediate(DestType._paramType);
+
         void* destData = _ParamImmediate(DestData)._pData;
 
         if (inputType->IsVariant()) {
             TypeRef variantInnerType = *reinterpret_cast<TypeRef *>_ParamImmediate(InputData._pData);
             if (variantInnerType && variantInnerType->IsA(destType, true)) {
-                variantInnerType->CopyData(variantInnerType->Begin(kPARead), destData);
+                if (destData) {
+                    variantInnerType->CopyData(variantInnerType->Begin(kPARead), destData);
+                }
             } else if (variantInnerType && destType->IsVariant()) {
-                *static_cast<TypeRef*>(destData) = CopyToVariant(variantInnerType);
+                if (destData) {
+                    *static_cast<TypeRef*>(destData) = CopyToVariant(variantInnerType);
+                }
             } else if (errPtr) {
                 SetVariantToDataTypeError(variantInnerType, destType, errPtr);
```

Comment:
We should set the destination data to default value for the type and add tests.

Reply:
Since this isn't directly related to what this PR intentions were, I will address this in a subsequent PR. I also created a work item to track this.